lib/Sema/TypeCheckDeclObjC.cpp

```diff
@@ -2534,16 +2543,28 @@ bool swift::diagnoseObjCMethodConflicts(SourceFile &sf) {
         if (auto accessor = dyn_cast<AccessorDecl>(originalMethod))
           originalDecl = accessor->getStorage();
 
+      // Conflicts between two imported ObjC methods are caused by the same
+      // clang decl having several Swift names.
+      if (originalDecl->hasClangNode() && conflictingDecl->hasClangNode())
+        continue;
```

Comment:
I wonder this is what was needed to handle the FIXME I left behind (and I think didn't file a bug for) in `test/Inputs/clang-importer-sdk/usr/include/EffectfulProperties.h` for this name clash on import as an async getter:

```
// FIXME: this does not raise an error, and generates two identical properties!
// -(void)anotherExampleWithCompletionBlock:(void (^)(NSString *))block
//   ASYNC_NAME("getter:manifestedString()");
// -(void)finalExampleWithReplyTo:(void (^)(NSString *))block
//   ASYNC_NAME("getter:manifestedString()");
```

Could you check if this gets resolved? Otherwise I'll file a proper bug for this.

Reply:
I doubt this change will affect that, because it affects how we diagnose two Swift methods with the same ObjC selector, whereas your example is two ObjC selectors with the same Swift name.